jxl/src/var_dct/dct.rs

```diff
@@ -108,20 +108,28 @@ define_idct_1d!(64, 32);
 define_idct_1d!(128, 64);
 define_idct_1d!(256, 128);
 
-fn transpose<const N: usize, const M: usize>(_data: &mut [f32]) {
-    todo!();
+fn transpose<const N: usize, const M: usize>(data: &[[f32; M]]) -> Vec<[f32; N]> {
+    let mut transposed: Vec<[f32; N]> = vec![[0.0f32; N]; M];
```

Comment:
In the future we almost certainly do not want this allocation here.

Reply:
changed to also use flat array here. Still doing one copy though.